src/metabase/query_processor/middleware/mark_outer_select_fields.clj

```diff
@@ -0,0 +1,33 @@
+(ns metabase.query-processor.middleware.mark-outer-select-fields
+  (:require [clojure.walk :as walk]))
+
+(defn- matching-fields?
+  [field source-field]
+  (every? #(= (field %) (source-field %)) [0 1]))
```

Comment:
Is matching the type and the "ID" of the clause good (enough)?

Reply:
It's really hard to parse this. I'm assuming `field` is a map representing a field. You are treating it like a vector, maybe a field_ref?

Reply:
My expectation was that since both `field` and `source-field` are direct children of a `:fields` clause they have the form `[:field name-or-id opts]` and I check if the first two respective elements match. I was hoping that I can use simple equality, but the `opts` can be different because middlewares can add different annotations based on where the fields are in the query. I'm not sure though that `opts` can fully be ignored (i.e., if nothing there is part of the "identity" of the field).

Reply:
ah yeah they are field-refs then. I have `metabase.query-processor.util/field-ref->key` that I use. I've seen @camsaul has a similar function somewhere that grabs what's important out of the options. If you reuse the function i've listed we can more easily keep track of all of these and fix them at once.